addons/editor-copy-scratchblocks/userscript.js

```diff
@@ -0,0 +1,58 @@
+// From https://github.com/apple502j/parse-sb3-blocks/releases/tag/v0.5.2
+import { toScratchblocks } from "./parse-sb3-blocks.module.js";
+
+// Some code referenced from blocks2image, block-switching and editor-devtools
+export default async function ({ addon, console, msg }) {
+  // Add right-click "Copy scratchblocks code" button to blocks
+  addon.tab.createBlockContextMenu(
+    (items, block) => {
+      if (addon.self.disabled) return items;
+      const makeSpaceItemIndex = items.findIndex((obj) => obj._isDevtoolsFirstItem);
+      const insertBeforeIndex =
+        makeSpaceItemIndex !== -1
+          ? // If "make space" button exists, add own items before it
+            makeSpaceItemIndex
+          : // If there's no such button, insert at end
+            items.length;
+      // "Copy scratchblocks code" message in /addons-l10n/en/editor-copy-scratchblocks.json (translate this)
+      items.splice(insertBeforeIndex, 0, {
+        enabled: true,
+        text: msg("copy-scratchblocks"),
+        callback: () => {
+          convertToScratchblocks(block);
+        },
+        separator: true,
+      });
+      return items;
+    },
+    { blocks: true }
+  );
+
+  async function convertToScratchblocks(block) {
+    // const blockId = block.getRootBlock().id; // Start from the topmost block, parsing the entire stack
+    let blockId = block.id;
+
+    // Get blocks in current sprite as project.json format
+    const vm = addon.tab.traps.vm;
+    let blocksJSON = vm.editingTarget.blocks._blocks;
```

Comment:
```suggestion
    let blocksJSON = JSON.parse(vm.toJSON(vm.editingTarget.id)).blocks;
```

Reply:
Thanks, that code works beautifully. I did some testing and it seems like this addon now works with any combination of blocks in vanilla Scratch. (parse-sb3-blocks would need code changes to support modded blocks/extensions if this addon gets put into TurboWarp or similar.)

Reply:
> (parse-sb3-blocks would need code changes to support modded blocks/extensions if this addon gets put into TurboWarp or similar.)

TurboWarp also serializes some vanilla blocks differently - see https://github.com/apple502j/parse-sb3-blocks/issues/9